### PR TITLE
Attachment Usages

### PR DESCRIPTION
**Pull request by deo002**
Project Details Attachment Usages Page

### CODE REVIEW

File: src/app/[locale]/projects/detail/[id]/components/AttachmentUsages.tsx [hunk 1]
```diff
@@ -272,6 +272,22 @@ export default function AttachmentUsages({ projectId }: { projectId: string }) {
                                     children: []
                                 }
                                 for(const att of r.attachments) {
+                                    let sourceCodeBundleChecked = false, manuallySetChecked = false, licenseInfoChecked = false
+                                    const usages = attachmentUsages["_embedded"]["sw360:attachmentUsages"]?.[0]?.filter((elem: any) => elem.attachmentContentId === att.attachmentContentId)
+                                    console.log(usages)
```
Comment (hoangnt2) [inline src/app/[locale]/projects/detail/[id]/components/AttachmentUsages.tsx:277]: remove log

File: src/app/[locale]/projects/detail/[id]/components/AttachmentUsages.tsx [hunk 2]
```diff
@@ -405,6 +424,21 @@ export default function AttachmentUsages({ projectId }: { projectId: string }) {
                         children: []
                     }
                     for(const att of r.attachments) {
+                        let sourceCodeBundleChecked = false, manuallySetChecked = false, licenseInfoChecked = false
```
Comment (hoangnt2) [inline src/app/[locale]/projects/detail/[id]/components/AttachmentUsages.tsx:427]: duplicated logic

**Review by heliocastro — APPROVED** [review]

File: src/app/[locale]/projects/detail/[id]/components/AttachmentUsages.tsx [hunk 3]
```diff
@@ -145,184 +208,87 @@ export default function AttachmentUsages({ projectId }: { projectId: string }) {
...
-                                proj[0]["_links"]["self"]["href"].lastIndexOf('/') + 1
-                            )}`}>
+                            <Link key={`${proj[0]["_links"]["self"]["href"].split('/').at(-1)}_name_project`} 
+                                href={`/projects/detail/${proj[0]["_links"]["self"]["href"].split('/').at(-1)}`}>
                                 {`${proj[0].name ?? ''} ${proj[0].version ?? ''}`}
                             </Link>,
-                            <div key={`${proj[0]["_links"]["self"]["href"].substring(
-                                proj[0]["_links"]["self"]["href"].lastIndexOf('/') + 1
-                            )}_relationship_project`}>
+                            <div key={`${proj[0]["_links"]["self"]["href"].split('/').at(-1)}_relationship_project`}>
```
Comment (hoangnt2) [inline src/app/[locale]/projects/detail/[id]/components/AttachmentUsages.tsx:221]: `${proj[0]["_links"]["self"]["href"].split('/').at(-1)}`
This code snippet is duplicated at many places, could you replace them by ${projectId}?

**Review by hoangnt2 — APPROVED** [review]
look good to me

**Pull request merged**